### PR TITLE
Adding chapter3 exercies

### PR DESCRIPTION
**Pull request by pxdmc**

### CODE REVIEW

File: chapter03/pxd/ex2.pl [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/perl -w
+
+use strict;
+use feature "say";
+
+my $dir = "/usr/lib/perl5/DataCash/*";
```
Comment (MarkWignall) [inline chapter03/pxd/ex2.pl]: You only use this in commented out code, it can be removed.

File: chapter03/pxd/ex2.pl [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+#opendir(D, "$dir") || die "Can't open directory $dir: $!\n";
+#my @files = readdir($dir);
+#closedir($dir);
+#my @files = glob $dir;
+
+my @files = </usr/lib/perl5/DataCash/*>;
```
Comment (MarkWignall) [inline chapter03/pxd/ex2.pl]: This seems like an interesting way to get the files.

File: chapter03/pxd/ex2.pl [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  exit 1;
+}
+
+say "Please enter a regex to match or type quit to exit: ";
+
+while ( chomp (  $user_regex = <STDIN> ) ) {
+
+  if ( !$user_regex || ($user_regex =~ /quit/  )) {
+	say "Please enter a regex to proceed...";
+	next;
```
Comment (MarkWignall) [inline chapter03/pxd/ex2.pl]: Should this not be last?

File: chapter03/pxd/ex2.pl [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+		push @hits, $file;
+	}
+   }
+  };
+  if ( $@ ) {
+    say "Regex is not valid. Try again! $@ $!";
+  } 
+
+  say "Matched files : " , map{$_,"\n"} @hits;
+  say "Please press enter to continue....";
```
Comment (MarkWignall) [inline chapter03/pxd/ex2.pl]: Don't they need to but in a regex first?

File: chapter03/pxd/ex2.pl (hunk 3 above)
Comment (pxdmc) [inline chapter03/pxd/ex2.pl]: True, I had initially just added the first check and extended it to quit and forgot about the `next` bit.

**Pull request merged**